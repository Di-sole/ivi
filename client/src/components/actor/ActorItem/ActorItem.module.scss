.container {
    display: block;
    max-width: 100%;

    &_large {
        max-width: 153px;
        @media (max-width: 1270px) {
            max-width: 181px;
        }
        @media (max-width: 1096px) {
            max-width: 188px;
        }
        @media (max-width: 920px) {
            max-width: 196px;
        }
        @media (max-width: 744px) {
            max-width: 211px;
        }
        @media (max-width: 600px) {
            max-width: 116px;
        }
        @media (max-width: 512px) {
            max-width: 114px;
        }
        @media (max-width: 392px) {
            max-width: 164px;
        }
    }
    &_medium {
        max-width: 128px;
        @media (max-width: 1159px) {
            max-width: 162.5px;
        }
        @media (max-width: 879px) {
            max-width: 185.75px;
        }
        @media (max-width: 599px) {
            max-width: 72px;
        }
    }

    &_small {
        max-width: 88px;
        @media (max-width: 599px) {
            max-width: 72px;
        }
    }

    .imageSection {
        position: relative;
        max-width: 100%;
        max-height: 100%;
        padding-top: 100%;

        &_large {
            width: 153px;
            @media (max-width: 1270px) {
                width: 181px;
            }
            @media (max-width: 1096px) {
                width: 188px;
            }
            @media (max-width: 920px) {
                width: 196px;
            }
            @media (max-width: 744px) {
                width: 211px;
            }
            @media (max-width: 600px) {
                width: 116px;
            }
            @media (max-width: 512px) {
                width: 144px;
            }
            @media (max-width: 392px) {
                width: 164px;
            }
        }

        &_medium {
            width: 128px;

            @media (max-width: 1159px) {
                width: 162.5px;
            }
            @media (max-width: 879px) {
                width: 185.75px;
            }
            @media (max-width: 599px) {
                width: 72px;
            }
        }

        &_small {
            width: 88px;

            @media (max-width: 599px) {
                width: 72px;
            }
        }

        .imageActorWrapper {
            position: absolute;
            top: 0;
            right: 0;

            border-radius: 50%;
            overflow: hidden;
            display: flex;
            align-items: center;
            width: 100%;
            height: 100%;
        }

        &_effect {
            transform-origin: bottom;
            transform: scale(1);
            transition-duration: 0.4s;
        }
    }

    .image {
        object-fit: cover;
        width: 100%;
        height: 100%;
    }

    .amountBadge {
        box-sizing: border-box;
        position: absolute;
        left: 4px;
        bottom: 0;
        background-color: #a869f0;
        height: 30px;
        padding: 5px 7.5px;
        border-radius: 2px;
        user-select: none;

        &__backBox {
            position: absolute;
            left: 2px;
            right: 2px;
            top: -2px;
            bottom: 100%;
            box-sizing: border-box;
            user-select: none;
            background-color: rgba(168, 105, 240, 0.64);
            border-top-left-radius: 1px;
            border-top-right-radius: 1px;
        }

        &__text {
            position: relative;
            text-align: center;

            color: #fff;
            font-size: 15px;
            line-height: 20px;
            font-weight: 800;
        }
    }

    .textSection {
        margin-top: 8px;

        &__title {
            color: #d9d7e0;
            font-weight: 600;

            white-space: nowrap;
            overflow: hidden;
            text-overflow: ellipsis;

            &_medium,
            &_large {
                width: 110px;
                line-height: 20px;
                font-size: 15px;
            }

            &_medium {
                @media (max-width: 599px) {
                    width: 65px;
                }
            }

            &_small {
                width: 80px;
                @media (max-width: 599px) {
                    width: 65px;
                }
                font-size: 13px;
                line-height: 16px;
            }
        }

        &__extra {
            color: #a5a1b2;
            font-size: 13px;
            line-height: 16px;
            font-weight: 500;

            white-space: nowrap;
            overflow: hidden;
            text-overflow: ellipsis;
            width: 80px;
            &_small {
                @media (max-width: 599px) {
                    width: 60px;
                }
            }
            &_small {
                @media (max-width: 599px) {
                    width: 65px;
                }
            }
        }
    }

    &:hover > .imageSection_effect {
        transition-duration: 0.4s;
        transform: scale(1.05);
    }

    &:hover > .textSection .textSection__title {
        color: #fff;
    }
}
