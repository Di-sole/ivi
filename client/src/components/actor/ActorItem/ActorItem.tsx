import Link from 'next/link';
import styles from './ActorItem.module.scss';
import Image from 'next/image';
import IActor from '../../../models/IActor';
import { declineWord } from '../../../utils/functions';

interface ActorItemProps {
    className?: string;
    actor: IActor;
    size: 'large' | 'medium' | 'small';
}

const ActorItem = ({ className, actor, size }: ActorItemProps) => {
    return (
        <Link href={`actors/${actor.id}`} className={[styles.container, className].join(' ')}>
            <div className={styles.container__imageContainer}>
                <div className={styles.container__imageWrapper}>
                    <Image
                        className={styles.container__image}
                        src={actor.img}
                        alt={`${actor.firstName} ${actor.lastName}`}
                        fill
                    />
                </div>

                {size === 'large' && <div className={styles.container__amountBadge}>{actor.amtMovies}</div>}
            </div>

            <div className={styles.container__textContainer}>
                <h2 className={styles.container__fullName}>
                    {actor.firstName}
                    <br />
                    {actor.lastName}
                </h2>

                {size === 'small' ? (
                    <h3 className={styles.container__role}>{actor.role}</h3>
                ) : (
                    <p className={styles.container__amountMovies}>
                        {actor.amtMovies} {declineWord(actor.amtMovies, ['фильм', 'фильма', 'фильмов'])}
                    </p>
                )}
            </div>
        </Link>
    );
};

export default ActorItem;
