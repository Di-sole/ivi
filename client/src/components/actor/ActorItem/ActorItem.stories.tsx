import { Meta, StoryObj } from '@storybook/react';
import ActorItem from './ActorItem';
import '../../../styles/nullstyle.scss';
import '../../../styles/globals.scss';
import './ActorItem.module.scss';

const meta: Meta<typeof ActorItem> = {
    title: 'Cards/Round',
    component: ActorItem,
    tags: ['autodocs'],
    argTypes: {
        size: {
            control: false,
        },
    },
};

export default meta;

type Story = StoryObj<typeof ActorItem>;

export interface IData {
    id: number;
    img: string;
    amtMovies: number;
    firstName: string;
    lastName: string;
    role: string;
}

const data: IData = {
    id: 1,
    img: require('../../../testAsserts/img/BG554460.jpg'),
    amtMovies: 4,
    firstName: 'Имя',
    lastName: 'Фамилия',
    role: 'Роль',
};

export const Effect: Story = {
    decorators: [
        (Story) => (
            <div style={{ width: '153px' }}>
                <Story />
            </div>
        ),
    ],
    args: {
        actor: data,
        size: 'large',
    },
};

export const Large: Story = {
    decorators: [
        (Story) => (
            <div style={{ width: '153px' }}>
                <Story />
            </div>
        ),
    ],
    args: {
        actor: data,
        size: 'large',
    },
};

export const Medium: Story = {
    decorators: [
        (Story) => (
            <div style={{ width: '128px' }}>
                <Story />
            </div>
        ),
    ],
    args: {
        actor: data,
        size: 'medium',
    },
};

export const Small: Story = {
    decorators: [
        (Story) => (
            <div style={{ width: '88px' }}>
                <Story />
            </div>
        ),
    ],
    args: {
        actor: data,
        size: 'small',
    },
};
