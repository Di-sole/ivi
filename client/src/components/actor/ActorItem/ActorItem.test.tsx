import { render, screen } from '@testing-library/react';
import ActorItem from './ActorItem';

export interface IData {
    id: number;
    img: string;
    amtMovies: number;
    firstName: string;
    lastName: string;
    role: string;
}

const data: IData = {
    id: 1,
    img: require('../../../testAsserts/img/BG554460.jpg'),
    amtMovies: 4,
    firstName: 'Имя',
    lastName: 'Фамилия',
    role: 'Роль',
};

describe('ACTORITEM SNAPSHOTS TESTS', () => {
    test('large ActorItem should not be changed', () => {
        render(<ActorItem actor={data} size="large" />);
        const link = screen.getByRole('link');
        expect(link).toMatchSnapshot();
    });

    test('medium ActorItem should not be changed', () => {
        render(<ActorItem actor={data} size="medium" />);
        const link = screen.getByRole('link');
        expect(link).toMatchSnapshot();
    });

    test('small ActorItem should not be changed', () => {
        render(<ActorItem actor={data} size="small" />);
        const link = screen.getByRole('link');
        expect(link).toMatchSnapshot();
    });
});
