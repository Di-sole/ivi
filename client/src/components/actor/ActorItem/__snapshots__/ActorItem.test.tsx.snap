// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ACTORITEM SNAPSHOTS TESTS large ActorItem should not be changed 1`] = `
<a
  class="container "
  href="actors/1"
>
  <div
    class="container__imageContainer"
  >
    <div
      class="container__imageWrapper"
    >
      <img
        alt="Имя Фамилия"
        class="container__image"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
    <div
      class="container__amountBadge"
    >
      4
    </div>
  </div>
  <div
    class="container__textContainer"
  >
    <h2
      class="container__fullName"
    >
      Имя
      <br />
      Фамилия
    </h2>
    <p
      class="container__amountMovies"
    >
      4
       
      фильма
    </p>
  </div>
</a>
`;

exports[`ACTORITEM SNAPSHOTS TESTS medium ActorItem should not be changed 1`] = `
<a
  class="container "
  href="actors/1"
>
  <div
    class="container__imageContainer"
  >
    <div
      class="container__imageWrapper"
    >
      <img
        alt="Имя Фамилия"
        class="container__image"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
  </div>
  <div
    class="container__textContainer"
  >
    <h2
      class="container__fullName"
    >
      Имя
      <br />
      Фамилия
    </h2>
    <p
      class="container__amountMovies"
    >
      4
       
      фильма
    </p>
  </div>
</a>
`;

exports[`ACTORITEM SNAPSHOTS TESTS small ActorItem should not be changed 1`] = `
<a
  class="container "
  href="actors/1"
>
  <div
    class="container__imageContainer"
  >
    <div
      class="container__imageWrapper"
    >
      <img
        alt="Имя Фамилия"
        class="container__image"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
  </div>
  <div
    class="container__textContainer"
  >
    <h2
      class="container__fullName"
    >
      Имя
      <br />
      Фамилия
    </h2>
    <h3
      class="container__role"
    >
      Роль
    </h3>
  </div>
</a>
`;
